### PR TITLE
feat: Move scheduling after booking

### PR DESCRIPTION
**Pull request by jkrvivian**
# Description of change

* Move the scheduler after the booker
* Remove FIFO-scheduler
* Avoid deleting discarded messages from the storage
  This causes panic when inheriting branches in the booker for its children.
* Avoid scheduling old messages (> 30 mins)
  To avoid exploding node queue sizes when syncing
* Make booker running on 1 single go-routine, since it is not thread-safe yet.

Data flow:
```
solidifier -> booker -> scheduler -> orderer -> gossip                                             
                     -> consensus -> tip manager
                     -> chat
                     -> approval weight
```

## Type of change

- Breaking change

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/tangle/scheduler.go [hunk 1]
```diff
@@ -104,7 +105,16 @@ func (s *Scheduler) Shutdown() {
 }
 
 // Setup sets up the behavior of the component by making it attach to the relevant events of the other components.
-func (s *Scheduler) Setup() {}
+func (s *Scheduler) Setup() {
+	// pass booked messages to the scheduler
+	s.tangle.Booker.Events.MessageBooked.Attach(events.NewClosure(func(messageID MessageID) {
+		if err := s.SubmitAndReady(messageID); err != nil {
+			s.Events.Error.Trigger(errors.Errorf("failed to submit to scheduler: %w", err))
```
Comment (Wollac) [inline packages/tangle/scheduler.go:112]: An error might be a bit strong here. There are 3 possibilities:

1. `ErrInboxExceeded` this is kind of expected behavior and we are triggering `NodeBlacklisted` anyways. So maybe this isn't even worth to log.
2. `ErrInsufficientMana` will never happen as the min access mana is sufficiently large.
3. Any other error coming from the `ObjectStorage` etc. These are probably actual errors.

File: packages/tangle/tangle.go [hunk 2]
```diff
@@ -52,7 +49,6 @@ type Tangle struct {
 	Events                *Events
 
 	setupParserOnce sync.Once
-	fifoScheduling  typeutils.AtomicBool
 	shutdownSignal  chan struct{}
```
Comment (Wollac) [inline packages/tangle/tangle.go:52]: Is `shutdownSignal` still used? It was introduced to cancel the `sleep` in the scheduler switching, which has been removed as part of this PR.
Of course, it also does not hurt either to have this in for future expansions.

**Comment by Wollac** [comment]
Does this new data-flow have any implications on the tip selection:
- Do we need to make sure to only select gossiped tips?
- Could this be causing issues right after syncing when nothing was gossiped?

File: packages/tangle/booker.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func NewBooker(tangle *Tangle) (messageBooker *Booker) {
 
 // Setup sets up the behavior of the component by making it attach to the relevant events of other components.
 func (b *Booker) Setup() {
-	b.tangle.Orderer.Events.MessageOrdered.Attach(events.NewClosure(func(messageID MessageID) {
+	b.tangle.Solidifier.Events.MessageSolid.Attach(events.NewClosure(func(messageID MessageID) {
```
Comment (capossele) [inline packages/tangle/booker.go:54]: As we typically have multiple go-routines that deal with solidification, wouldn't these go-routines then also continue the execution of the booking?
I think the flow should be interrupted here and have the booker running with only 1 single go-routine. WDYT?

File: packages/tangle/tangle.go [hunk 4]
```diff
@@ -125,6 +117,13 @@ func (t *Tangle) Setup() {
 	t.ConsensusManager.Setup()
 	t.TipManager.Setup()
 
+	// increase scheduler rate if the node is not synced at start
+	if !t.Synced() {
+		rate := t.Scheduler.Rate()
+		rate -= rate / 2 // 50% increase
+		t.Scheduler.SetRate(rate)
+	}
```
Comment (capossele) [inline packages/tangle/tangle.go:125]: Considering that we bypass the scheduler for all the messages older than 5 minutes, do we still need this at all?

**Review by capossele — APPROVED** [review]
Just a minor change/suggestions, otherwise seems solid

File: packages/tangle/tangle.go (hunk 4 above)
Comment (jkrvivian) [reply packages/tangle/tangle.go:125]: Let's remove it!

**Pull request merged**